src/helpers/extensionStatus.js

```diff
@@ -1,55 +1,60 @@
 import { useState, useEffect } from 'react'
-import get from 'lodash/get'
 import { useFlag } from 'cozy-flags'
 
 export const extensionStatuses = {
   checking: 'checking',
   installed: 'installed',
-  notInstalled: 'not-installed'
+  notInstalled: 'not-installed',
+  connected: 'connected'
 }
 
 export const useExtensionStatus = () => {
-  const [installed, setInstalled] = useState(extensionStatuses.notInstalled)
+  const [status, setStatus] = useState(extensionStatuses.notInstalled)
   const extensionCheckDisabled = useFlag('extension-check-disabled')
 
   useEffect(() => {
     if (extensionCheckDisabled) {
       return
     }
 
-    const checkExtensionInstallation = () => {
-      window.postMessage(
-        {
-          message: { source: 'cozy-passwords' }
-        },
-        // This star means that we are sending the message to all
-        // tabs / extensions. This is not ideal, but since we don't know the
-        // cozy extension URI, we can't target it directly
-        '*'
-      )
+    const checkExtensionStatus = () => {
+      const event = document.createEvent('Event')
```

Comment:
Why was window.postMessage abandonned ? (naive question)

Reply:
There's some details in https://github.com/cozy/cozy-passwords/pull/16/commits/08f00479a52682f76b8b7d6115cdf98b8b0ddf37 commit message. But my next task is to write a documentation about how the communication between the app and the extension.

Reply:
One big advantage of dropping postMessage is that we don't have this `*` sending the message to all tabs/extensions anymore